### PR TITLE
Add GaussianNoise

### LINKED ISSUE #6192: Add Gaussian noise transformation
### 🚀 The feature

Add gaussian noise transformation in the functionalities of torchvision.transforms.

### Motivation, pitch

Using Normalizing Flows, is good to add some light noise in the inputs. Right now I am using albumentation for this but, would be great to use it in the torchvision library

### Alternatives

[Albumentation has a gaussian noise implementation](https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.GaussianBlur)

### Additional context

_No response_

cc @vfdev-5 @datumbox

### PR DESCRIPTION
**Pull request by Richienb**
Fixes #6192
Fixes #6233

cc @vfdev-5

### CODE REVIEW

**Review by wink94 — APPROVED** [review]

File: torchvision/transforms/v2/_misc.py [hunk 1]
```diff
@@ -205,6 +205,31 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
...
+    The input tensor is also expected to be of float dtype in ``[0, 1]``.
+    This transform does not support PIL images.
+
+    Args:
+        mean (float): Mean of the sampled normal distribution. Default is 0.
+        sigma (float): Standard deviation of the sampled normal distribution. Default is 0.1.
+        clip (bool, optional): Whether to clip the values in ``[0, 1]`` after adding noise. Default is True.
+    """
+
+    def __init__(self, mean: float = 0.0, sigma: float = 0.1, clip=True) -> None:
```
Comment (Richienb) [inline torchvision/transforms/v2/_misc.py:223]: I think `sigma` (σ) should instead be called `var` for "variance". For example, `mean` is not called `mu` (μ).

Comment (NicolasHug) [reply torchvision/transforms/v2/_misc.py:223]: That's fair @Richienb , I can't claim `mean` and `sigma` are the best name combination, but unfortunately for consistency this is probably what we'll need to use, as these are the names that are already used by `GaussianBlur`.

Comment (Richienb) [reply torchvision/transforms/v2/_misc.py:223]: Ok, let's save this for a future breaking change.

**Review by NicolasHug — APPROVED** [review]
Thanks for the PR @Richienb , as you've seen I've made a few changes to the original implementation, but this LGTM now. In particular the transform enforces tensors with float dtype (PIL isn't supported). We might add support for that later but that would require a dtype conversion to avoid over/underflows, so it's best to treat it separately. Thank you again for the PR!

**Pull request merged**